src/librustc_resolve/diagnostics.rs

```diff
@@ -627,6 +627,9 @@ trait Foo {
         Self; // error: unresolved name `Self`
     }
 }
+
+// or:
+let x = unknown_variable;  // error: unresolved name `unknown_variable`
 ```
 
 Please verify you didn't misspell the name or that you're not using an
```

Comment:
Please verify that the name wasn't misspelled and ensure that the identifier being referred to is valid for the given situation.